app/design/frontend/base/default/template/somethingdigital/criteo/basket.phtml

```diff
@@ -2,15 +2,32 @@
 
 <?php
     echo $this->getChildHtml('somethingdigital_criteo_device_detection');
+    $email = Mage::helper('somethingdigital_criteo')->getHashedEmail();
+    $items = $this->getCriteoViewBasket();
+    $useDataLayer = Mage::getStoreConfig('somethingdigital_criteo/settings/use_datalayer');
 ?>
-
-<script type="text/javascript">
-    window.criteo_q = window.criteo_q || [];
-    window.criteo_q.push(
-        { event: "setAccount", account: "<?php echo Mage::getStoreConfig('somethingdigital_criteo/settings/account_number'); ?>" },
-        { event: "setHashedEmail", email: ["<?php echo Mage::helper('somethingdigital_criteo')->getHashedEmail();?>"] },
-        { event: "setSiteType", type: deviceType },
-        { event: "viewBasket", item: <?php echo $this->getCriteoViewBasket(); ?> }
-    );
-</script>
-<?php endif; ?>
\ No newline at end of file
+    <?php if (!$useDataLayer): ?>
+        <script type="text/javascript">
+            window.criteo_q = window.criteo_q || [];
+            window.criteo_q.push(
+                { event: "setAccount", account: "<?php echo Mage::getStoreConfig('somethingdigital_criteo/settings/account_number'); ?>" },
+                { event: "setHashedEmail", email: ["<?php echo $email;?>"] },
+                { event: "setSiteType", type: deviceType },
+                { event: "viewBasket", item: <?php echo $items ?> }
+            );
+        </script>
+    <?php else: ?>
+        <script type="text/javascript">
+            var dataLayer = dataLayer || [];
+            dataLayer.push(
+                {
+                    event: "crto_basketpage",
```

Comment:
I'm not following the purpose of the event? It seems this is typically used for HTML DOM events (https://developers.google.com/tag-manager/devguide#events) which this is not. Am I missing something?

Reply:
This event is via the criteo guide [here](http://guides.criteotilt.com/tag-manager/gtm/?accountid=52244&advertiserid=47350&advertisername=Baked+by+Melissa&utm_medium=52244&homepageurl=http%3A%2F%2Fwww.bakedbymelissa.com%2F&accuratekey=bakedbymelissaus%3B7815412%3B7815413%3B52244)

Reply:
Was not aware you were following a guide. I think it's worth including a link to the guide in your implementation for explanation purposes. Perhaps it could be in the `<comment>` for the setting?

```
When enabled, the information (like email, product list etc) is pushed to the datalayer rather than sending these to the criteo endpoint. Implementation is based on <a href="http://guides.criteotilt.com/tag-manager/gtm/">Criteo documentation</a>.
```